crates/agent/src/api/handlers.rs

```diff
@@ -441,3 +442,130 @@ pub async fn cleanup_cache(
         message: format!("Cleaned up {} expired cache entries", cleaned_entries),
     }))
 }
+
+/// Register a new user
+#[utoipa::path(
+    post,
+    path = "/auth/register",
+    request_body = CreateUserRequest,
+    responses(
+        (status = 201, description = "User registered successfully", body = RegisterResponse),
+        (status = 400, description = "Invalid request data"),
+        (status = 409, description = "User already exists"),
+        (status = 500, description = "Internal server error")
+    ),
+    tag = "Authentication",
+    summary = "Register a new user",
+    description = descriptions::HANDLER_USER_REGISTRATION_DESCRIPTION
+)]
+pub async fn register(
+    State(state): State<AppState>,
+    Json(request): Json<CreateUserRequest>,
+) -> Result<Json<RegisterResponse>, ApiError> {
+    let response = state.auth_service.register(request).await.map_err(|e| {
+        error!("Error registering user: {:?}", e);
+        match e {
+            crate::utils::error::Error::Validation(msg) => ApiError::bad_request(msg),
+            _ => ApiError::internal_error(format!("Failed to register user: {}", e)),
+        }
+    })?;
+
+    Ok(Json(response))
+}
+
+/// Login user
+#[utoipa::path(
+    post,
+    path = "/auth/login",
+    request_body = LoginRequest,
+    responses(
+        (status = 200, description = "Login successful", body = LoginResponse),
+        (status = 400, description = "Invalid credentials"),
+        (status = 500, description = "Internal server error")
+    ),
+    tag = "Authentication",
+    summary = "Login user",
+    description = descriptions::HANDLER_USER_LOGIN_DESCRIPTION
+)]
+pub async fn login(
+    State(state): State<AppState>,
+    Json(request): Json<LoginRequest>,
+) -> Result<Json<LoginResponse>, ApiError> {
+    let response = state.auth_service.login(request).await.map_err(|e| {
+        error!("Error logging in user: {:?}", e);
+        match e {
+            crate::utils::error::Error::Validation(msg) => ApiError::bad_request(msg),
+            _ => ApiError::internal_error(format!("Failed to login user: {}", e)),
+        }
+    })?;
+
+    Ok(Json(response))
+}
+
+/// Get current user profile
+#[utoipa::path(
+    get,
+    path = "/auth/me",
+    responses(
+        (status = 200, description = "User profile retrieved successfully", body = UserProfile),
+        (status = 401, description = "Unauthorized"),
+        (status = 500, description = "Internal server error")
+    ),
+    tag = "Authentication",
+    summary = "Get current user profile",
+    description = descriptions::HANDLER_GET_CURRENT_USER_PROFILE_DESCRIPTION
+)]
+pub async fn get_current_user_profile(
```

Comment:
Here, we should add the authentication tag for Swagger as the jwt token is required